core/trino-main/src/test/java/io/trino/execution/TestDropColumnTask.java

```diff
@@ -0,0 +1,184 @@
+
+        getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("b"), false, false));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("a", BIGINT)));
+    }
+
+    @Test
+    public void testDropOnlyColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("test", BIGINT)));
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("test"), false, false)))
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessageContaining(" Cannot drop the only column in a table");
+    }
+
+    @Test
+    public void testDroColumnNotExistingTable()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("not_existing_table");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("test"), false, false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", tableName);
+    }
+
+    @Test
+    public void testDropColumnNotExistingTableIfExists()
+    {
+        QualifiedName tableName = qualifiedName("not_existing_table");
+
+        getFutureValue(executeDropColumn(tableName, QualifiedName.of("test"), true, false));
+        // no exception
+    }
+
+    @Test
+    public void testDroColumnNotExistingColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("not_existing"), false, false)))
+                .hasErrorCode(COLUMN_NOT_FOUND)
+                .hasMessageContaining("Column 'not_existing' does not exist");
+    }
+
+    @Test
+    public void testDropColumnIfExists()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+
+        getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("c"), false, true));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("a", BIGINT), new ColumnMetadata("b", BIGINT)));
+    }
+
+    @Test
+    public void testUnsupportedDropDuplicatedField()
```

Comment:
Do we have a similar test in `AddColumn` with a duplicate field ?

Reply:
`AddColumn` doesn't support adding fields yet. I can add a test for adding a column with duplicated fields with row type, but the test purpose will differ.

Reply:
Maybe we could test for adding duplicate columns